frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx

```diff
@@ -123,15 +117,19 @@ export const ParameterSettings = ({
   const isMultiValue = getIsMultiSelect(parameter) ? "multi" : "single";
 
   const handleTypeChange = (sectionId: string) => {
-    // can be moved to redux?
-    const defaultOptionOfNextType = defaultOptionForSection[sectionId];
+    const option = defaultOptionForSection[sectionId];
 
-    onChangeType(defaultOptionOfNextType.type, sectionId);
+    onChangeType(option);
   };
 
   const handleOperatorChange = (operatorType: string) => {
-    // TODO: fix sectionId type
-    onChangeType(operatorType, sectionId as string);
+    const option = parameterSections
```

Comment:
not optimal, so I created a followup https://github.com/metabase/metabase/issues/41517